parity-clib/src/lib.rs

```diff
@@ -163,3 +170,95 @@ impl CallbackStr {
 		}
 	}
 }
+
+#[cfg(feature = "jni")]
+#[no_mangle]
+pub unsafe extern fn Java_io_parity_ethereum_Parity_configFromCli(env: JNIEnv, _: JClass, cli: jobjectArray) -> jlong {
```

Comment:
I'm not a `JNI` expect, but `extern "system"` is used in the [example](https://github.com/prevoty/jni-rs/blob/master/example/mylib/src/lib.rs#L26), see https://github.com/prevoty/jni-rs/pull/22.